### DISCUSSION TITLE
Is there a way to have music typeset using Quarto?

### DISCUSSION THREAD

**Discussion by TS-CUBED** [quarto-dev/quarto-cli#7098] [Q&A]
### Description

There are a few markup languages for setting music (e.g. ABCnotation), which would be perfect in combination with a markdown based system like Quarto.

Are there any plans to integrate something like this? In a similar way as mermaid and graphviz integration works at the moment.

**Comment by mcanouil** [comment]
No concrete plans to integrate this in a foreseeable future but the extension is there especially for this.
See for example <https://github.com/dfalbel/chord-sheet>

See the documentation: <https://quarto.org/docs/extensions/>.

Note that there is an opened issue for this, but you better work on an extension that wait for this to arrive in Quarto as there are other priorities.

- https://github.com/quarto-dev/quarto-cli/issues/2700

**Comment by TS-CUBED** [comment]
Thanks for confirming that it's not planned. Pity.

The extension mentioned is for guitar chords only and not for music typesetting.

I guess one could write an extension for ABC in a similar way (create image from ABC notation?)

Edit: answered to the email notification, which didn't have the note about the open issue in it. I hadn't found that (did search for music, so not sure why that wasn't found). Will look at that.

**Reply by mcanouil** [reply]
I know the extension I shared is for guitar it's kind of in the name, but you want the same kind of extension thus it is a good example accompanied with Quarto documentation.

**Reply by TS-CUBED** [reply]
Apologies if that came across wrong. I just wanted to add that for documentation. I will have a look and see if the chord-sheet extension can be adapted to use ABC.js instead of chordSheetJS

**Comment by TS-CUBED** [comment]
Just for the record in case somebody is looking for a quick way to make this work for HTML documents.

Using ABC.js (https://paulrosen.github.io/abcjs/) this can be achieved with just an `include-in-header` and a plain HTML block `{=html}`:

````
---
title: 'Test for ABC include'

license: "CC BY"

format:
  html:
    include-in-header:
      - text: <script src="https://cdn.jsdelivr.net/npm/abcjs/dist/abcjs-plugin-min.js" type="text/javascript"></script>
---

```{=html}
X: 1
T: Cooley's
M: 4/4
L: 1/8
K: Emin
|:D2|"Em"EB{c}BA B2 EB|~B2 AB dBAG|"D"FDAD BDAD|FDAD dAFD|
"Em"EBBA B2 EB|B2 AB defg|"D"afe^c dBAF|"Em"DEFD E2:|
|:gf|"Em"eB B2 efge|eB B2 gedB|"D"A2 FA DAFA|A2 FA defg|
"Em"eB B2 eBgB|eB B2 defg|"D"afe^c dBAF|"Em"DEFD E2:|
```
````

With the abc-plugin, the ABC code is just entered directly in the HTML file, so there's no HTML code needed. Does only work for HTML output, of course.

**Comment by nichtich** [comment]
I am looking for [lilypond](https://lilypond.org/) integration.

**Reply by chillinflute** [reply]
This is obviously a long time past, but there's a lua filter for this: https://github.com/pandoc/lua-filters/tree/master/lilypond

**Comment by chillinflute** [comment]
For anyone still seeking this kind of functionality, a slightly more elegant solution can be found in the [Musify](https://github.com/sebastiandelbano/musify) extension, which I just discovered and have found to be working quite well so far for my needs. It also uses ABC via `abc.js` and will display the code, notes, and a playback option. Works with HTML only (although PDF rendering seems like a viable development for the extension for those with the coding chops).

**Reply by mcanouil** [reply]
It's on my extensions listing:

- https://m.canouil.dev/quarto-extensions/?q=Musify
